internal/service/alarms/api/openapi.yaml

```diff
@@ -0,0 +1,756 @@
+            format: uuid
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AlarmEventRecordModifications'
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AlarmEventRecordModifications'
+        '404':
+          description: Not found
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Internal server error
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+  /O2ims_infrastructureMonitoring/v1/alarmSubscriptions:
+    get:
+      operationId: GetSubscriptions
+      summary: Retrieve the list of alarm subscriptions
+      tags:
+        - subscriptions
+      parameters:
+        - in: query
+          name: filter
+          schema:
+            type: string
+          allowReserved: true
+          description: Optional filter parameter. Only the attributes in AlarmSubscriptionInfo are filterable.
+          example: (eq,callback,http://my.callback.com)
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/AlarmSubscriptionInfo'
+        '400':
+          description: Bad request
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Internal server error
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+    post:
+      operationId: CreateSubscription
+      summary: Create a new alarm subscription
+      tags:
+        - subscriptions
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AlarmSubscriptionInfo'
+      responses:
+        '201':
+          description: Successful creation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AlarmSubscriptionInfo'
+        '400':
+          description: Bad request
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Internal server error
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+      callbacks:
```

Comment:
This is not part of this API. We will need to generate a client for this

Reply:
Yeah basically the idea is that `CreateSubscription` will generate a callback depending on `subscriber.callbackUrl` as seen below. 

The codegen gives us `AlarmEventNotification` go struct. We just need to create Post call using the callback url.

Reply:
I've seen these callbacks defined separately, so that we can share it with customers who are gonna generate the server code. From the same spec we could generate the client (and the go struct we need). My point is to have these things separated since they don't belong to the same server.

Reply:
As discussed on call...CreateSubscription is the "owner" of the callback.